packages/backend/src/createL2Clients.ts

```diff
@@ -0,0 +1,38 @@
+import { Layer2Rpc } from '@l2beat/config'
+import { providers } from 'ethers'
+
+import { Config } from './config'
+import { EthereumClient } from './peripherals/ethereum/EthereumClient'
+
+export function createL2Clients(config: Config) {
```

Comment:
I think I would structure this a bit differently. We create the clients specifically so that we can get their transactions by creating and starting RpcTransactionUpdater. If we are creating the clients in a separate step then when we introduce new types of transaction updaters we'd need to somehow also differentiate the clients making the code quite a bit more complicated.

I suggest the following:
1. Create `/src/setup` that will be the go to directory for extracting logic from `Application` and move this file there
2. Instead of returning EthereumClients create the clients and then immediately create RpcTransactionUpdater. That way when new types of configuration show up you just handle them differently in the if statement
3. Instead of mapping and filtering just create the updaters array and conditionally push to it (this is just a style preference, but I think the code would be more readable)
4. Then Application just receives an array of updates instead of having to create it by itself